MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/mas/OptimizationActivity.java

```diff
@@ -0,0 +1,211 @@
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_mas_optimization);
+
+    //Set up stop list
+    stops = new ArrayList<Position>();
+
+    // Set First Stop
+    final Position origin = Position.fromCoordinates(-122.408818, 37.784015);
+    stops.add(origin);
+
+    // Setup the MapView
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+        map = mapboxMap;
+
+        // Add origin and destination to the map
+        mapboxMap.addMarker(new MarkerOptions()
+          .position(new LatLng(origin.getLatitude(), origin.getLongitude()))
+          .title("1"));
+
+        map.setOnMapClickListener(new MapboxMap.OnMapClickListener() {
+          @Override
+          public void onMapClick(@NonNull LatLng point) {
+
+            // Optimization API is limited to 12 coordinate sets
+            if (stops.size() == 12) {
+              Toast.makeText(OptimizationActivity.this, "Only 12 stops allowed on route", Toast.LENGTH_LONG).show();
+            } else {
+              map.addMarker(new MarkerOptions()
+                .position(new LatLng(point.getLatitude(), point.getLongitude()))
+                .title("1"));
+
+              stops.add(Position.fromCoordinates(point.getLongitude(), point.getLatitude()));
+
+              getOptimized(stops);
+            }
+          }
+        });
+      }
+    });
+  }
+
+  private void getOptimized(List<Position> coordinates) {
+
+    optimizedClient = new MapboxOptimizedTrips.Builder()
+      .setSource("first")
+      .setDestination("any")
+      .setCoordinates(coordinates)
+      .setOverview(DirectionsCriteria.OVERVIEW_FULL)
+      .setProfile(DirectionsCriteria.PROFILE_DRIVING)
+      .setAccessToken(Mapbox.getAccessToken())
+      .build();
+
+    optimizedClient.enqueueCall(new Callback<OptimizedTripsResponse>() {
+      @Override
+      public void onResponse(Call<OptimizedTripsResponse> call, Response<OptimizedTripsResponse> response) {
+        System.out.println(call.request().url().toString());
+
+        // You can get the generic HTTP info about the response
+        Log.d(TAG, "Response code: " + response.body().getTrips());
+        Log.d(TAG, "response: " + call.request().url().toString());
+
+        if (response.body() == null) {
+          Log.e(TAG, "No routes found, make sure you set the right user and access token.");
+          return;
+        } else if (response.body().getTrips().size() < 1) {
```

Comment:
In order to make the statement more semantic, why not using `response.body().getTrips().isEmpty()` instead?